execution/stepExecutor.go

```diff
@@ -45,7 +45,9 @@ func (e *stepExecutor) executeStep(step *gauge.Step, protoStep *gauge_messages.P
 	if !stepResult.GetFailed() {
 		executeStepMessage := &gauge_messages.Message{MessageType: gauge_messages.Message_ExecuteStep, ExecuteStepRequest: stepRequest}
 		stepExecutionStatus := e.runner.ExecuteAndGetStatus(executeStepMessage)
+		e.currentExecutionInfo.Stacktrace = ""
 		if stepExecutionStatus.GetFailed() {
+			e.currentExecutionInfo.Stacktrace = stepExecutionStatus.GetStackTrace()
```

Comment:
Using `e.currentExecutionInfo.Stacktrace` probably is not a good idea because it does not belong to any entity(`step`, `scenario` or `spec`). We might remove this field in future.

If you want to assign stacktrace for `step` then we can add `stacktrace` field to `e.currentExecutionInfo.CurrentStep` and assign when there is a `step` failure.

For this, you will have to change [gauge-proto](https://github.com/getgauge/gauge-proto/) and add a new field `stacktrace` in [`StepInfo`](https://github.com/getgauge/gauge-proto/blob/master/messages.proto#L112). After that you can use it in `gauge` and `gauge-java`.